### PR TITLE
feat: create sentry-expo config plugin

### PR DESCRIPTION
**Pull request by cruzach**
configure native projects automatically on eject and when building managed projects on EAS Build

### CODE REVIEW

**Comment by byCedric** [comment]
Looks awesome, bit weird why CI is saying `dist/sentry.expo.js` is outdated. I don't think you changed that right?

**Comment by cruzach** [comment]
it was a problem on master, fixed here - https://github.com/expo/sentry-expo/commit/19abb5dfb1891ffb9341d1c10de0269b6748d05f 

definitely my fault, thank goodness for that CI check 🙏

File: src/plugin/withSentryIOS.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    );
+  script.shellScript = JSON.stringify(code);
+}
+
+export function writeSentryPropertiesTo(filepath: string, sentryProperties: string) {
+  if (!fs.existsSync(filepath)) {
+    throw new Error("Directory '" + filepath + "' does not exist.");
+  }
+
+  fs.writeFileSync(path.resolve(filepath, 'sentry.properties'), sentryProperties);
```
Comment (EvanBacon) [inline src/plugin/withSentryIOS.ts:69]: Does this file need to be added to the xcode project?

File: src/plugin/withSentryAndroid.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      writeSentryPropertiesTo(
+        path.resolve(config.modRequest.projectRoot, 'android'),
+        sentryProperties
+      );
+      return config;
+    },
+  ]);
+};
+
+function modifyBuildGradle(buildGradle: string) {
```
Comment (EvanBacon) [inline src/plugin/withSentryAndroid.ts:29]:
```suggestion
function modifyAppBuildGradle(buildGradle: string) {
```
makes it easier to follow along. Maybe also add a quick doc-block that says "writes to the app/build.gradle" for ppl who are trying to quickly understand the setup.

**Review by EvanBacon — CHANGES_REQUESTED** [review]
I'd feel more comfortable if this plugin used the correct tsconfig for transpilation. Could you open another PR that migrates this package to expo-module-scripts, merge it, then move the plugin to a plugin folder. 

Further please add some tests for `getSentryProperties`, `modifyAppBuildGradle`, `modifyExistingXcodeBuildScript`. The tests can be tested using `yarn test plugin` after the package has been migrated to EMS.

File: src/plugin/withSentryIOS.ts (hunk 1 above)
Comment (cruzach) [reply src/plugin/withSentryIOS.ts:69]: It's just used in one of the build-phase shell scripts to grab properties, so it shouldn't need to be added to Xcode 

sentry's own scripts for adding this via their "wizard" tool (which is a sick name btw) don't add this to the xcode project files

**Review by EvanBacon — APPROVED** [review]
LGTM

**Pull request merged**